File: frontend/src/app/models/comment.ts
import {User} from "./user";
import {Post} from "./post";

export class Comment {
  id: number;
  text: string;
  date: string;
  user: User = new User();
  post: Post = new Post();

  constructor(text?: string, userId?: number, postId?: number) {
    this.text = text;
    this.user.id = userId;
    this.post.id = postId;
  }
}

File: fapi/src/main/java/com/netcracker/fapi/entity/Comment.java
package com.netcracker.fapi.entity;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.time.LocalDateTime;

public class Comment {
    private Long id;
    private String text;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy HH:mm:ss")
    private LocalDateTime date;
    private Post post;
    private User user;

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }
}
